### PR TITLE
Consolidate ticket_early_data_info and early_data into a single

### PR DESCRIPTION
**Pull request by ekr**
extension, as suggested by Steven Valdez/David Benjamin.

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -2444,13 +2446,20 @@ supply an "early_data" extension as well as the "pre_shared_key"
...
 
 %%% Key Exchange Messages
 
        struct {
+           select (Handshake.msg_type) {
+               case client_hello:
+               case hello_retry_request:
+               case new_session_ticket:
+                  uint32 max_early_data_size;
        } EarlyDataIndication;
```
Comment (kazu-yamamoto) [inline draft-ietf-tls-tls13.md]: Is "}" missing?

Comment (kazu-yamamoto) [inline draft-ietf-tls-tls13.md]: This is fall through. So, `uint32` is contained in `client_hello`. But the original extension is empty. How should we use `uint32` for `client_hello`?

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -2454,6 +2454,7 @@ The "extension_data" field of this extension contains an
            select (Handshake.msg_type) {
                case client_hello:
                case hello_retry_request:
+                  // empty.
```
Comment (kazu-yamamoto) [inline draft-ietf-tls-tls13.md]: According to our old discussion, I think that `struct {}` is correct.

Comment (ekr) [inline draft-ietf-tls-tls13.md]: Ugh. I re-ordered instead.

**Comment by kaduk** [comment]
LGTM

**Pull request merged**